base/reflection.jl

```diff
@@ -1375,22 +1387,11 @@ It also supports the following syntax:
 - `@invokelatest xs[i]` expands to `Base.invokelatest(getindex, xs, i)`
 - `@invokelatest xs[i] = v` expands to `Base.invokelatest(setindex!, xs, v, i)`
 
-```jldoctest
-julia> @macroexpand @invokelatest f(x; kw=kwv)
-:(Base.invokelatest(f, x; kw = kwv))
-
-julia> @macroexpand @invokelatest x.f
-:(Base.invokelatest(Base.getproperty, x, :f))
-
-julia> @macroexpand @invokelatest x.f = v
-:(Base.invokelatest(Base.setproperty!, x, :f, v))
-
-julia> @macroexpand @invokelatest xs[i]
-:(Base.invokelatest(Base.getindex, xs, i))
-
-julia> @macroexpand @invokelatest xs[i] = v
-:(Base.invokelatest(Base.setindex!, xs, v, i))
-```
+!!! note
+    If `f` is a global, it will be resolved consistently
+    in the (latest) world as the call target. However, all other arguments
+    (as well as `f` itself if it is not a literal global) will be evaluated
+    in the current world age.
```

Comment:
Does this mean that in `invokelatest(f, args...)`, `f` itself is not evaluated in the latest world, whereas using `@invokelatest f(args...)` ensures that both the function binding and the function call are evaluated in the latest world?

I’ve written the following snippet to find the difference, but it seems that `invokelatest` behaves as expected (i.e. always returns latest result). I’m curious to know in what situations the difference between these two would become significant?
```julia
julia> function foo()
           global fg3 = sin
           x = invokelatest(fg3, 42)
           Main.fg3 = cos
           y = invokelatest(fg3, 42)
           x, y
       end
foo (generic function with 1 method)

julia> foo()
(-0.9165215479156338, -0.39998531498835127)
```